backends/p4tools/p4tools.def

```diff
@@ -261,3 +261,31 @@ class HeaderExpression : StructExpression {
         BUG_CHECK(structType == nullptr || structType->is<IR::Type_Header>(), "%1%: unexpected header type", structType->node_type_name());
     }
 }
+
+/// An wrapper which models an InOut argument.
+/// Contains both a copy of the original reference as well as the resolved value of that reference.
+class InOutReference : Expression {
+/// Because we do not use a reference we need to construct the value normally.
```

Comment:
This is a limitation of the `.def` files. `ref` is not a pointer of reference, but a full member. But this does not work well with how the `irgenerator` interprets `.def` files and the constructor that is generated is incorrect. We need to write our own constructor in this case.

Reply:
Hmm, I did not notice it is inline. I guess that makes sense for `StateVariable`. Maybe extend the comment a little? Something like:

```
/// Because the ref member is inline the IR generator does not generate a reasonable constructor. Therefore, we have to write the constructors manually.